freetypefont.mod/freetypefont.bmx

```diff
@@ -67,6 +68,10 @@ Type TFreeTypeGlyph Extends TGlyph
 		h=_h
 	End Method
 
+	Method Index:Int() Override

```

Comment:
I think it should be preferred to extend the original "TGlyph" to contain the `field _index:int`. Why? Because when using the HBFont-Stuff and `DrawText()` this will call `Method Index:Int()` each draw-call - for each character in the text.

Especially for "debug builds" this will create a lot of debugscope (enter, leave) and for "release builds" I am not sure if the GCC optimizes it out (the method call) at the end or not. Yes, having an extra field for "basic" `TGlyph` is not "clean" but I guess in this situation it will be a cood compromise to not sacrifice some performance (Font rendering with single glyph-images is already ... slow).

Reply:
Tested the performance difference but I guess the actual rendering is way more expensive so this can be neglected ... for release builds!

sample code: https://github.com/bmx-ng/text.mod/pull/30#issuecomment-2271064987
release:
```
index():
1000000x HBImageFont.DrawText took 8696ms.
1000000x ImageFont.DrawText took 6309ms.

_index:
1000000x HBImageFont.DrawText took 8621ms.
1000000x ImageFont.DrawText took 6272ms.
```

(cannot test debug for now - see raised issue - but I think it will have a visible impact there)